src/common/buffer.cc

```diff
@@ -984,6 +984,7 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     char* ptr = _raw->data + _off + _len;
     *ptr = c;
     _len++;
+    assert(_len != 0);
```

Comment:
I'd prefer if these asserts are compiled conditionally, for full debug builds only. This is already performance sensitive path that affects everyone.